src/daft-scan/src/lib.rs

```diff
@@ -93,6 +101,7 @@ pub struct ScanTask {
     // TODO(Clark): Directly use the Pushdowns struct as part of the ScanTask struct?
     pub columns: Option<Arc<Vec<String>>>,
     pub limit: Option<usize>,
+    pub size_bytes_file: Option<u64>,
```

Comment:
Nit: `size_bytes_files` plural instead?

Should we just recompute from `sources` when queried, instead of caching? Might result in fewer bugs down the line when we start doing slicing/coalescing of `ScanTasks`

Reply:
For the naming, this was supposed to be `size_bytes_file` vs `size_bytes_memory`, to juxtapose against the table statistics which roll up an in-memory estimate, but maybe this can be a bit more clear. How about `size_bytes_on_file` or something like that?

As for whether we should cache, I'm thinking that this should follow the same pattern was table stats and metadata, where the cached data is mergable on a union, and needs to be recomputed on a split, so I don't foresee too many bugs there! And although we call this `O(1)` times per scan task now, I wouldn't be surprised if we end up hitting this more frequently in a scheduling loop when doing memory backpressure judgements and the like (we had to do something similar to this for Ray Data).

But if you feel strongly about this, happy to get rid of this caching!